### PR TITLE
Fix bugs in object file engine

### PR DESCRIPTION
**Pull request by MarkMcCaskey**
Still fixing some bugs, work in progress

# Review

- [ ] Add a short description of the the change to the CHANGELOG.md file

### CODE REVIEW

**Review by syrusakbary — APPROVED** [review]

File: lib/vm/src/libcalls.rs [hunk 1]
```diff
@@ -464,6 +464,11 @@ impl LibCall {
             Self::FloorF64 => "wasmer_f64_floor",
             Self::NearestF32 => "wasmer_f32_nearest",
             Self::NearestF64 => "wasmer_f64_nearest",
+            // We have to do this because mac requires a leading `_` and it's not
```
Comment (Hywan) [inline lib/vm/src/libcalls.rs:467]:
```suggestion
            // We have to do this because macOS requires a leading `_` and it's not
```

**Review by Hywan — APPROVED** [review]

**Review by syrusakbary — APPROVED** [review]

**Comment by syrusakbary** [comment]
I successfully tested `wasm-validate.wasm` and `rustpython.wasm`

**Pull request merged**